### PR TITLE
API V2 Update

### PR DESCRIPTION
**Pull request by asleepingpanda**
Updates the Web App to use the CF Apollos 2.0 API in preparation for release.

### CODE REVIEW

**Comment by rkdavidson** [comment]
Looks like the Netlify build is having trouble generating static pages for `/events` etc with the new API.

I can smoketest this branch locally later today, but there's a warning about some fetch `encoding` dependency missing that may be a clue.

The requests are all getting 400's, so feels like we're not properly formatting things for the server.

image

**Comment by rkdavidson** [comment]
Hey @asleepingpanda let us know if you need help resolving the conflicts. It'd still be great to get this merged today or tomorrow, so we can be spoke testing as we wrap up development odds and ends this cycle.

**Comment by rkdavidson** [comment]
Hey @asleepingpanda , is this PR obsolete now that the Apollos 2.0 update landed in `master`?

**Comment by asleepingpanda** [comment]
@rkdavidson that's no bueno... I'll look into it with Danny today. This branch is def still "alive" and I'm sure there will be plenty of conflicts to work through so we'll do another smoke test of it today

**Review by jordanwade — DISMISSED** [review]
Probably could kill this empty file. 
image

Routing is broken, it might be related to to all the ServerSideProps stuff being commented out.

Seems like this PR has ballooned up to include things that weren't entirely necessary to  'fix' API v2 issues, I could be wrong. In the future it would probably make it easier to review if PRs are as surgical as possible. Also, it makes it so there is less guess work on what is causing bugs.

File: pages/[title].js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  return (
+    <Layout title={capitalize(title)}>
+      <FeatureFeedProvider Component={FeatureFeed} options={options} />
+    </Layout>
+  );
+}
+
+// export async function getServerSideProps() {
```
[the remark this answers is not in the document]
Comment (asleepingpanda) [reply pages/[title].js:34]: Yeah good catch. Let me test that real quick. It's just a quick copy-paste error on my end.

File: ui-kit/ThemeProvider/ThemeProvider.js [hunk 2]
```diff
@@ -16,7 +17,7 @@ function ThemeProvider(props = {}) {
   };
 
   return (
-    <SCThemeProvider theme={theme}>
+    <SCThemeProvider theme={merge(theme, props?.themeMixin || {})}>
```
[the remark this answers is not in the document]
Comment (asleepingpanda) [reply ui-kit/ThemeProvider/ThemeProvider.js:20]: I _think_ this was some random OG code when we were all talking on a call once that accidentally got merged. It's not actually being used and it'll for sure just get overwritten by whatever your PR is that solves this problem.

**Comment by asleepingpanda** [comment]
> Probably could kill this empty file.
> image
> 
> Routing is broken, it might be related to to all the ServerSideProps stuff being commented out.
> 
> Seems like this PR has ballooned up to include things that weren't entirely necessary to 'fix' API v2 issues, I could be wrong. In the future it would probably make it easier to review if PRs are as surgical as possible. Also, it makes it so there is less guess work on what is causing bugs.

Agreed that this got away from us and there won't be another PR like this in the future. Awkward migration/upgrade stuff caused this to get away from us cause our tickets were too vague. 

I pushed out a fix for the messages routing error.

File: pages/[title].js (hunk 1 above)
[the remark this answers is not in the document]
Comment (asleepingpanda) [reply pages/[title].js:34]: Looks ok on my end. Could you try it again?

**Review by dzwood — DISMISSED** [review]
Looks good to me, Feature Feeds are loading from the Apollos 2.0 branch on the API. Let's merge and start on the other branches :)

**Review by rkdavidson — DISMISSED** [review]

**Review by rkdavidson — APPROVED** [review]
Haven't tested the recent changes locally and still a bit uneasy about the scope here, but confident we can tackle any new odds and ends early next week.

**Pull request merged**